index.js

```diff
@@ -119,4 +139,15 @@ Cloneable.isCloneable = function (stream) {
   return stream instanceof Cloneable || stream instanceof Clone
 }
 
+Clone.prototype._destroy = function (err, cb) {
+  if (!err) {
+    this.push(null)
+    this.end()
```

Comment:
i'd get rid of the push(null) and end() calls. this makes error handling *super* tricky as the stream will look like it ended gracefully (unless that's what you are going for of course)

Reply:
Are you sure? We are doing these only if there was no error.

Reply:
Depends on your usecase. i'd *never* do it. Plenty of things do .destroy() to just teardown the pipeline without emitting errors

Reply:
This module is relying heavily on that behavior to work properly. Unfortunately it's a massive rewrite and this fix an important bug.